### PR TITLE
Bugfix lpdev 181 invalid key name

### PR DESCRIPTION
**Pull request by tarurar**

### CODE REVIEW

File: src/Lykke.Service.PaySign.Services/KeyStoreService.cs [hunk 1]
```diff
@@ -1 +1,41 @@
...
+        private readonly Dictionary<string, KeyInfo> _keyInfos;
+
+        public KeyStoreService()
+        {
+            _keyInfos = new Dictionary<string, KeyInfo>();
+        }
+
+        public void Add(string name, KeyInfo keyInfo)
+        {
+            lock (_keyInfos)
```
Comment (MikhailGerasimov) [inline src/Lykke.Service.PaySign.Services/KeyStoreService.cs]: The lock looks here suspicious - the only write method protected. I would used ConcurrentDictionary here.

File: src/Lykke.Service.PaySign/Controllers/KeysController.cs [hunk 2]
```diff
@@ -81,6 +82,13 @@ public class KeysController : Controller
 
                 return NoContent();
             }
+            catch (KeyAlreadyExistsException keyEx)
+            {
+                await _log.WriteErrorAsync(nameof(KeysController), nameof(UploadKey), new {keyEx.KeyName}.ToJson(),
```
Comment (MikhailGerasimov) [inline src/Lykke.Service.PaySign/Controllers/KeysController.cs]: Consider using sync version of logger

File: src/Lykke.Service.PaySign/Controllers/SignController.cs [hunk 3]
```diff
@@ -31,6 +34,7 @@ public class SignController : Controller
         [HttpPost("{keyName}")]
         [SwaggerOperation("Sign")]
         [ProducesResponseType(typeof(string), (int) HttpStatusCode.OK)]
+        [ProducesResponseType(typeof(ErrorResponse), (int)HttpStatusCode.BadRequest)]
         [ProducesResponseType(typeof(void), (int) HttpStatusCode.InternalServerError)]
         public async Task<IActionResult> Sign(string keyName, [FromBody] string content)
         {
```
Comment (MikhailGerasimov) [inline src/Lykke.Service.PaySign/Controllers/SignController.cs]: Validate incoming arguments here

File: src/Lykke.Service.PaySign/Controllers/KeysController.cs [hunk 4]
```diff
@@ -81,6 +82,13 @@ public class KeysController : Controller
 
                 return NoContent();
             }
+            catch (KeyAlreadyExistsException keyEx)
```
Comment (MikhailGerasimov) [inline src/Lykke.Service.PaySign/Controllers/KeysController.cs]: Validate incoming arguments

**Review by MikhailGerasimov — COMMENTED** [review]
Update https://github.com/LykkeCity/Lykke.Service.PaySign/blob/0c24ff9cb7a41431088ee31023c8b6ccadadd95d/src/Lykke.Service.PaySign/Dockerfile
The example: https://github.com/LykkeCity/lykke.dotnettemplates/blob/master/Lykke.Service.LykkeService/src/Lykke.Service.LykkeService/Dockerfile#L1

**Review by polonzer — APPROVED** [review]

**Pull request merged**